left sidebar: Add Views label for spectators.

opmkumar opened the pull request:
This PR adds the Views label in the left side bar for the spectators without it being collapsible.

Fixes #30324 

**Screenshots and screen captures:**

![image](https://github.com/zulip/zulip/assets/142340063/890a9bd4-8c85-4be4-bd46-26e454157e85)


<details>

<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on web/src/left_sidebar_navigation_area.ts:
```diff
@@ -130,6 +130,12 @@ export function handle_narrow_activated(filter: Filter): void {
+    if (page_params.is_spectator) {
+        // If the user is a spectator, do nothing
+        return;
+    }
```
Is this code necessary? I would have thought it would be unreachable with the button being hidden...

opmkumar replied:
Yes, I think it is necessary. I have removed the hidden-for-spectator class from the div with id "views-label-container" so that my leftbar-side-title ("Views") would be visible to spectators and further to prevent the toggling of the view-label-container it was necessary to add the above written code. Since the previously written code was already handling what needs to be hidden from the spectators (like inbox, mentions, etc.) in the expanded view the removal of hidden-for-spectator from views-label-container shouldn't be a problem and toggling wont be possible due to the added code in the ts file.

opmkumar replied:
@timabbott can you please review my approach and provide me your views on it?

timabbott replied:
OK, this does seem correct to not have clicking "VIEWS" do something, but we should also disable `cursor: pointer` on that element for spectators.

As a side note, I think you could have made to product impact a lot more clear in your answer. Let's make the comment say something explaining the why "// We don't support collapsing VIEWS for spectators, so exit early."

alya commented:
Works for me in manual testing!

@timabbott please take a look at the reply to your question above.

opmkumar commented:
@timabbott I have removed the ```cursor: pointer``` and added ```cursor: default``` to the Views label as you have told based on some conditional styling (whether user is a spectator or not)i n the views-label-container in left_sidebar.hbs file.

Also I have changed the comment suggested by you in the ts file.
Kindly review my code and provide me your views on it.

timabbott commented:
Looks great; I removed a stray newline and rewrote the commit message to have a bit more explanation, and marked this to merge once CI passes. Thanks for the contribution @opmkumar!
